src/main/java/seedu/algobase/logic/parser/FindCommandParser.java

```diff
@@ -16,18 +27,70 @@
     /**
      * Parses the given {@code String} of arguments in the context of the FindCommand
      * and returns a FindCommand object for execution.
+     *
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        String trimmedArgs = args.trim();
-        if (trimmedArgs.isEmpty()) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argumentMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_AUTHOR, PREFIX_DESCRIPTION, PREFIX_SOURCE,
+                        PREFIX_DIFFICULTY, PREFIX_TAG);
+
+        // According to the command format, no preamble should be present.
+        if (!argumentMultimap.getPreamble().isBlank()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
         }
 
-        String[] nameKeywords = trimmedArgs.split("\\s+");
+        FindCommand.FindProblemDescriptor findProblemDescriptor = new FindCommand.FindProblemDescriptor();
 
-        return new FindCommand(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
-    }
+        if (argumentMultimap.getValue(PREFIX_NAME).isPresent()) {
+            String trimmedNameArg = argumentMultimap.getValue(PREFIX_NAME).get().trim();
+            String[] nameKeywords = trimmedNameArg.split("\\s+");
+            findProblemDescriptor.setNamePredicate(new NameContainsKeywordsPredicate(Arrays.asList(nameKeywords)));
+        }
+
+        if (argumentMultimap.getValue(PREFIX_AUTHOR).isPresent()) {
+            String authorKeyword = argumentMultimap.getValue(PREFIX_AUTHOR).get();
+            findProblemDescriptor.setAuthorPredicate(new AuthorMatchesKeywordPredicate(authorKeyword));
+        }
+
+        if (argumentMultimap.getValue(PREFIX_DESCRIPTION).isPresent()) {
+            String trimmedDescriptionArg = argumentMultimap.getValue(PREFIX_DESCRIPTION).get();
+            String[] descriptionKeywords = trimmedDescriptionArg.split("\\s+");
+            findProblemDescriptor.setDescriptionPredicate(
+                new DescriptionContainsKeywordsPredicate(Arrays.asList(descriptionKeywords)));
+        }
+
+        if (argumentMultimap.getValue(PREFIX_SOURCE).isPresent()) {
+            String sourceKeyword = argumentMultimap.getValue(PREFIX_SOURCE).get();
+            findProblemDescriptor.setSourcePredicate(new SourceMatchesKeywordPredicate(sourceKeyword));
+        }
+
+        if (argumentMultimap.getValue(PREFIX_DIFFICULTY).isPresent()) {
+            String[] difficultyBounds = argumentMultimap.getValue(PREFIX_DIFFICULTY).get().split("-");
+            if (difficultyBounds.length != 2) {
```

Comment:
Can define a method that checks if an array has a given length and throws an exception otherwise